transformations/discourse_marker_substitution/transformation.py

```diff
@@ -0,0 +1,142 @@
+import random
+import re
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+"""
+Substitution of discourse markers with semantically equivalent marker
+"""
+
+CLASS_TO_MARKERS = {
+    "Comparison.Contrast": ["but", "however", "on the contrary"],
+    "Comparison.Contrast.Juxtaposition": ["by comparison", "whereas"],
+    "Contingency.Cause.Reason": ["because", "as", "since", "inasmuch as"],
+    "Contingency.Cause.Result": [
+        "so",
+        "as a result",
+        "consequently",
+        "thus",
+        "therefore",
+        "accordingly",
+        "hence",
+    ],
+    "Expansion.Alternative.Chosen alternative": ["instead"],
+    "Expansion.Alternative.Conjunctive": ["or"],
+    "Expansion.Conjunction": [
+        "and",
+        "in fact",
+        "while",
+        "also",
+        "furthermore",
+        "in addition",
+        "meanwhile",
+        "moreover",
+        "similarly",
+        "further",
+        "in turn",
+        "likewise",
+        "besides",
+        "additionally",
+    ],
+    "Expansion.Instantiation": ["for example", "for instance"],
+    "Expansion.Restatement.Equivalence": ["in other words"],
+    "Expansion.Restatement.Generalization": ["in sum"],
+    "Expansion.Restatement.Specification": ["specifically", "in particular"],
+    "Temporal.Asynchronous.Precedence": ["then", "subsequently", "finally", "later"],
+    "Temporal.Asynchronous.Succession": ["previously", "earlier", "after"],
+    "Temporal.Synchrony": ["at the time", "when", "at the same time"],
+}
+
+MARKER_TO_CLASS = {
+    "accordingly,": "Contingency.Cause.Result",
+    "additionally,": "Expansion.Conjunction",
```

Comment:
I think the comma could be removed and you could ensure that you match correctly with sentences with and without a comma.

Reply:
Hi! Thank you for your review. The comma reduces the recall but increases the precision. You can find some cases where no comma leads to undesirable matchings.